### PR TITLE
[GEARPUMP-339] Add ScalaDoc to window api and impl

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [ ] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

**Comment by manuzhang** [comment]
R: @huafengw

**Comment by huafengw** [comment]
No unit test failed, so where does the failure come from?

**Comment by manuzhang** [comment]
Maybe it's killed after hanging since it took longer than usual.

File: streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/impl/WindowRunner.scala [hunk 1]
```diff
@@ -137,11 +156,14 @@ class DefaultWindowRunner[IN, OUT](
...
         } else {
-          // minimum of end of last triggered window and start of first un-triggered window
+          // The output watermark is the minimum of end of last triggered window
+          // and start of first un-triggered window
           TriggeredOutputs(outputs, TaskUtil.min(wmk, firstWin.startTime))
         }
       } else {
+        // All windows have been triggered.
         if (time == Watermark.MAX) {
+          // This means there will no more inputs so it's safe to advance to the maximum watermark.
```
Comment (huafengw) [inline streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/impl/WindowRunner.scala]: there will be no more

Comment (manuzhang) [inline streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/impl/WindowRunner.scala]: nice catch 😏

**Comment by manuzhang** [comment]
BTW, the tests passes locally and this PR doesn't touch any codes.

**Comment by huafengw** [comment]
+1

**Pull request closed without merging**